recipe-server/client/control_new/state/app/recipes/reducers.js

```diff
@@ -38,23 +38,30 @@ function history(state = new Map(), action) {
   }
 }
 
+const formatRecipe = recipe =>
+  recipe
+    .set('action_id', recipe.getIn(['action', 'id'], null))
+    .set('latest_revision_id', recipe.getIn(['latest_revision', 'id'], null))
+    .set('approved_revision_id', recipe.getIn(['approved_revision', 'id'], null))
+    .remove('action')
+    .remove('latest_revision')
+    .remove('approved_revision');
 
 function items(state = new Map(), action) {
-  let recipe;
-
   switch (action.type) {
     case RECIPE_RECEIVE: {
-      recipe = fromJS(action.recipe);
+      const recipe = fromJS(action.recipe);
+      return state.set(action.recipe.id, formatRecipe(recipe));
+    }
 
-      recipe = recipe
-        .set('action_id', recipe.getIn(['action', 'id'], null))
-        .set('latest_revision_id', recipe.getIn(['latest_revision', 'id'], null))
-        .set('approved_revision_id', recipe.getIn(['approved_revision', 'id'], null))
-        .remove('action')
-        .remove('latest_revision')
-        .remove('approved_revision');
+    case RECIPE_PAGE_RECEIVE: {
+      const recipes = fromJS(action.recipes.results);
 
-      return state.set(action.recipe.id, recipe);
+      return state.withMutations(mutState => {
```

Comment:
I think this is way weirder syntax. It's harder to read to me.

I was under the impression the `withMutations` stuff was going to apply to the `formatRecipe` and `formatRevision` methods.

Reply:
This is what I had in mind. What do you find harder to read about it?

Applying this only to the `formatRecipe` and `formatRevision` functions would have some of the performance  benefits, but wrapping the entire reducer transaction in `withMutation` (like done here) is better.